src/middleware/inject_params.rs

```diff
@@ -59,7 +65,7 @@ impl Middleware<CallRequest, Result<JsonValue, Error>> for InjectParamsMiddlewar
                 // without current block
                 let to_inject = self.get_parameter().await;
                 tracing::debug!("Injected param {} to method {}", &to_inject, request.method);
-                request.params.push(to_inject);
+                request.params.insert(idx, to_inject);
```

Comment:
this could fail for methods with multiple optional parameters. i.e. `state_getKeysPaged`. in that case we need to inject null for other missing params